libraries/chain/include/eosio/chain/abi_serializer.hpp

```diff
@@ -81,12 +85,30 @@ struct abi_serializer {
       return false;
    }
 
-   typedef std::function<fc::variant(fc::datastream<const char*>&, bool, bool, const fc::time_point&)>  unpack_function;
-   typedef std::function<void(const fc::variant&, fc::datastream<char*>&, bool, bool, const fc::time_point&)>  pack_function;
+   typedef std::function<fc::variant(fc::datastream<const char*>&, bool, bool, const abi_serializer::yield_function_t&)>  unpack_function;
+   typedef std::function<void(const fc::variant&, fc::datastream<char*>&, bool, bool, const abi_serializer::yield_function_t&)>  pack_function;
 
    void add_specialized_unpack_pack( const string& name, std::pair<abi_serializer::unpack_function, abi_serializer::pack_function> unpack_pack );
 
-   static const size_t max_recursion_depth = 32; // arbitrary depth to prevent infinite recursion
+   static constexpr size_t max_recursion_depth = 32; // arbitrary depth to prevent infinite recursion
+
+   // create standard yield function that checks for max_serialization_time and max_recursion_depth.
+   // now() deadline caputered at time of this call
+   static yield_function_t create_yield_function(const fc::microseconds& max_serialization_time) {
+      fc::time_point deadline = fc::time_point::now();
+      if( max_serialization_time > fc::microseconds::maximum() - deadline.time_since_epoch() ) {
+         deadline = fc::time_point::maximum();
+      } else {
+         deadline += max_serialization_time;
+      }
+      return [max_serialization_time, deadline](size_t recursion_depth) {
```

Comment:
const size_t recursion_depth

Reply:
I'm not a fan of const integral types in function parameters, seems like extra syntax for no gain.